rpm: require mandatory signature

Cause an error rather than warning if a package only has NOKEY or
NOTTRUSTED signature.

Upstream-Status: Inappropriate [wrlinux specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>

Modify to apply to current version of package.c.

Signed-off-by: Joe Slater <joe.slater@windriver.com>

Modify to apply to current version of package.c.
Use sinfo->rc instead of the undeclared rc.

Signed-off-by: Chen Qi <Qi.Chen@windriver.com>
---
 lib/package.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/lib/package.c b/lib/package.c
index 081123d..73bdf53 100644
--- a/lib/package.c
+++ b/lib/package.c
@@ -278,7 +278,12 @@ static int handlePkgVS(struct rpmsinfo_s *sinfo, void *cbdata)
     case RPMRC_NOKEY:		/* Public key is unavailable. */
 	/* XXX Print NOKEY/NOTTRUSTED warning only once. */
 	if (stashKeyid(sinfo->keyid) == 0)
+#ifdef MANDATORY_KNOWN_SIG
+	    lvl = RPMLOG_ERR;
+	    sinfo->rc = RPMRC_FAIL;
+#else
 	    lvl = RPMLOG_WARNING;
+#endif
 	break;
     case RPMRC_NOTFOUND:	/* Signature/digest not present. */
 	lvl = RPMLOG_WARNING;
-- 
2.7.4

